build.zig

```diff
@@ -0,0 +1,259 @@
+const std = @import("std");
+const Build = std.Build;
+const Step = Build.Step;
+const Compile = Step.Compile;
+const Dependency = Build.Dependency;
+const fs = std.fs;
+
+pub fn build(b: *Build) !void {
+    const target = b.standardTargetOptions(.{});
+    const optimize = b.standardOptimizeOption(.{});
+
+    const libsodium_dep = b.dependency(
+        "libsodium",
+        .{ .target = target, .optimize = optimize, .static = true, .shared = false },
+    );
+    const libsodium = libsodium_dep.artifact("sodium");
+
+    const win_build = target.result.isMinGW();
+    var winpthreads_dep: ?*Dependency = null;
+    var winpthreads: ?*Compile = null;
+    if (win_build) {
+        winpthreads_dep = b.dependency(
+            "winpthreads",
+            .{ .target = target, .optimize = optimize },
+        );
+        winpthreads =
+            winpthreads_dep.?.artifact("winpthreads");
+    }
+    const gt_dep = b.dependency("gtest", .{ .target = target, .optimize = optimize });
+    const cmp_dep = b.dependency("cmp", .{});
+    // we copy the third_party dependencies into the source tree
+    // not an ideal solution but we need them there and git
+    // submodule want work for zig dependency snapshots.
+    // we do the copy only if the file does not exist at the
+    // target position
+    const copy_files = b.addWriteFiles();
+    inline for (.{ "cmp.h", "cmp.c" }) |file| {
+        const rpath = b.pathJoin(&.{ "third_party/cmp", file });
+        const apath = b.pathFromRoot(rpath);
+        if (fs.accessAbsolute(apath, .{})) |_| {} else |_| {
+            // if the file does not exist we copy it
+            _ = copy_files.addCopyFileToSource(
+                .{ .dependency = .{ .dependency = cmp_dep, .sub_path = file } },
+                rpath,
+            );
+        }
+    }
+
+    const lib_src_files = &.{
```

Comment:
Can we extract this list from the cmake code? I'd like to address nurupo's very valid concern by making this build system as easy to maintain as any other ci check. E.g. the conanfile is kind of a build system wrapper around cmake. If we can make the zig one a wrapper too, I think it would be less of a concern.

Reply:
This means parsing of Makefile.inc and use the files in the libtoxcore_la_SOURCES variable. 
And we drop the gtest stuff for now? For the downstream zig use they are not important anyway.
But then maybe we should use a fixed zig version for CI because a new zig could require the change of build.zig and this would be a bourdon for other maintainers.